src/thor/astar.cc

```diff
@@ -410,22 +413,24 @@ void AStarPathAlgorithm::SetOrigin(GraphReader& graphreader,
     // destination is in a forward direction along the edge. Add back in
     // the edge score/penalty to account for destination edges farther from
     // the input location lat,lon.
-    auto p = destinations_.find(edgeid);
-    if (p != destinations_.end()) {
+    auto settled_dest_edge = destinations_percent_along_.find(edgeid);
+    if (settled_dest_edge != destinations_percent_along_.end()) {
       if (IsTrivial(edgeid, origin, destination)) {
         // Find the destination edge and update cost.
-        for (const auto& destination_edge : destination.path_edges()) {
-          if (destination_edge.graph_id() == edgeid) {
+        for (const auto& dest_path_edge : destination.path_edges()) {
+          if (dest_path_edge.graph_id() == edgeid) {
```

Comment:
i think there is a simplification here. im pretty sure `IsTrivial` is superfluous and the extra condition it adds  can be added to the `if` on line 421. dont feel like you have to do it, im just pointing it out in case you want to simplify this

Reply:
:+1: I'll leave it as is for now.